### ISSUE TITLE
Language Pragmas Auto-Insert After Module Docs 

### ISSUE THREAD

**Issue by emilypi** [haskell/haskell-language-server#1726]
### Your environment
Output of `haskell-language-server --probe-tools` or `haskell-language-server-wrapper --probe-tools`:

```sh
haskell-language-server version: 1.1.0.0 (GHC: 8.10.4) (PATH: /Users/emilypi/.ghcup/bin/haskell-language-server-wrapper-1.1.0) (GIT hash: f1c096927186a93d8e3ccd4fe8385cc1b070350b)
Tool versions found on the $PATH
cabal:		3.4.0.0
stack:		2.5.1
ghc:		8.10.4
```
Which OS do you use:
Big Sur OSX

Which lsp-client do you use:
Emacs lsp-haskell
Describe your project (alternative: link to the project):
Cabal project
Contents of `hie.yaml`:
```yaml
cradle:
  cabal:
    - path: "Cabal/src"
      component: "lib:Cabal"

    - path: "cabal-install/src"
      component: "lib:cabal-install"

    - path: "cabal-install/main/Main.hs"
      component: "cabal-install:exe:cabal"

    - path: "cabal-install/tests"
      component: "cabal-install:test:unit-tests"

    - path: "cabal-install/tests"
      component: "cabal-install:test:memory-usage-tests"

    - path: "cabal-install/tests"
      component: "cabal-install:test:integration-tests2"

    - path: "cabal-install/tests"
      component: "cabal-install:test:long-tests"

    - path: "cabal-install-solver/src"
      component: "lib:cabal-install-solver"

    - path: "cabal-install-solver/src-assertion"
      component: "lib:cabal-install-solver"
```

### Steps to reproduce

On HLS 1.1, auto-insert any language pragma in a module with header documentation

### Expected behaviour
Pragmas should be inserted before docs

### Actual behaviour
Pragmas are inserted *after* docs: 

```haskell
{-# LANGUAGE BangPatterns #-}
{-# LANGUAGE DeriveFunctor #-}
{-# LANGUAGE DeriveFoldable #-}
{-# LANGUAGE DeriveTraversable #-}
-- | Cabal-like file AST types: 'Field', 'Section' etc,
--
-- This (intermediate) data type is used for pretty-printing.
--
-- @since 3.0.0.0
--
{-# LANGUAGE LambdaCase #-}
module Distribution.Fields.Pretty 
```

**Comment by OliverMadine** [comment]
I believe that the intended behaviour is to insert pragmas directly the start of the code, since the following code finds the first non-blank line and inserts the pragma directly above.

```Haskell
endOfModuleHeader :: ParsedModule -> Range
endOfModuleHeader pm =
  let mod = unLoc $ pm_parsed_source pm
      modNameLoc = getLoc <$> hsmodName mod
      firstImportLoc = getLoc <$> listToMaybe (hsmodImports mod)
      firstDeclLoc = getLoc <$> listToMaybe (hsmodDecls mod)
      line = maybe 0 (_line . _start) (modNameLoc <|> firstImportLoc <|> firstDeclLoc >>= srcSpanToRange)
      loc = Position line 0
   in Range loc loc
```

We could include an option to always insert pragmas at the top of the file? I will open a PR for this shortly.

**Comment by jneira** [comment]
I am afraid there was a pr that remove just that behaviour cause it put the pragma before she-bangs #1340 which fixed #555
I think the ideal behaviour should be identify the code span of the last pragmas block (as we could have several pragmas blocks with comments between them) and insert the new one just after it
